docs/adr/adr-00XX-making-the-component-theme-customisable.md

```diff
@@ -0,0 +1,251 @@
+---
+kind: '📌 Docs/Architecture Decision Records'
+---
+
+# ADR 00XX: Making the component theme customisable
+
+🗓️ 2022-XX-XX · ✍️ Florian Sanders
+
+## The context
+
+As explained in the ADR #0017 - Finding accessible colors and creating Design Tokens, we introduced design tokens divided into two categories:
+
+* "choices" that represent the colors used by our components. They are supposed to be private, not exposed and not customizable. For instance `--color-red: #be242d`.
+* "decisions" that represent our semantic choices applying specific colors for specific content types. These are supposed to be public and customizable. For instance `--color-text-success`.
+
+These tokens were stored as CSS Custom Props, inside a JavaScript file that needed to be imported by components.
+
+The goal of our semantic tokens, the "decisions", is that they can be used as some kind of API for theme customisation.
+If one wanted to customise the common theme (mainly the colors at the moment) of our components, they would insert their own colors as values of these semantic tokens.
+For instance:
+```css
+--color-text-success: your value;
+```
+
+## The problems
+
+There were several limitations that made properly customising the theme impossible.
+These limitations were actually well known because the first step was building the design system before making it customisable.
+
+### Serving and using the theme
+
+The theme was contained in a JavaScript file and every component had to import it.
+
+```js
+import { defaultThemeStyles } from '../styles/default-theme.js';
+...
+static get styles () {
+  return [
+    defaultTheme,
+    css`
+      ...
+    `,
+  ],
+}
+```
+
+This meant that our tokens were processed as a constructable stylesheet and it was very difficult to override their values from a CSS file outside the components.
+
+Keeping the JavaScript file to provide our tokens would have meant relying on the build process to allow people to customise the tokens.
+Since we also make our components available via CDN, we did not want to rely on the build process to allow people to override our token values.
+
+### Planning for a dark theme
+
+Our components had issues with background colors that would block theme customisation:
+
+* some of them did not define a default background color. This did not cause any issue as long as the context in which they were integrated used a white color background, but as soon as you changed the background color of their parent, some issues appeared.
+* some of them had a background color hard coded. If you wanted to use the components in a dark mode, changing our `color-bg-default` token would have no effect and the component backgrounds would remain white. On the other hand, their default font color was inherited. Thus, you could end up with a white text color, inherited from your website dark mode colors, over a white background color, hard coded within the components.
```

Comment:
@pdesoyres-cc do you find this section easier to read ?
I'm not sure it's any better :joy:

Reply:
I had to read it twice :joy: but this is because it is hard to explain!
Anyway, I really find it better like that! :+1: